lab_3_generate_by_ngrams/main.py

```diff
@@ -236,6 +365,11 @@ def _extract_n_grams(
 
         In case of corrupt input arguments, None is returned
         """
+        if not isinstance(self._encoded_corpus, tuple) or self._encoded_corpus is None:
```

Comment:
Why `self`? We're working here with arguments i believe

And if this is a check for empty encoded corpus, then I don't think it'll work...